java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -166,13 +168,23 @@ public void setInitialCapacity(int numRecords) {
    *                This helps in tightly controlling the memory we provision
    *                for inner data vector.
    */
+  @Override
   public void setInitialCapacity(int numRecords, double density) {
+    if ((numRecords * density) >= 2_000_000_000) {
+      throw new OversizedAllocationException("Requested amount of memory is more than max allowed");
```

Comment:
If possible can we add some context here like the current capacity control variables? Useful in debugging.

Reply:
Sure, I will file a JIRA for this since this sort of problem needs to be addressed throughout the code.